myia/prim/py_implementations.py

```diff
@@ -571,6 +572,10 @@ def broadcast_shape(shpx, shpy):
             shp.append(b)
         elif b == 1:
             shp.append(a)
+        elif a == ANYTHING:
+            shp.append(b)
+        elif b == ANYTHING:
+            shp.append(a)
```

Comment:
I'm not sure about this.  ANYTHING should never be encountered during actual execution so it feels out of place here.

Reply:
Yeah, that's a valid point. It's a code reuse thing: it is useful for the inferrer to perform this operation on ANYTHING, and during execution, well, these paths won't be entered anyway. Do you think it would be better to duplicate this functionality?

Reply:
On one hand I don't like duplicating and on the other hand this also feels wrong.  So I'm not sure what the perfect solution here.

We might as well go with this for now, it shouldn't be super hard to split it up later.